modules/nf-scil/registration/tractogram/main.nf

```diff
@@ -0,0 +1,83 @@
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
+        'scilus/scilus:1.6.0' }"
+
+    input:
+    tuple val(meta), path(anat), path(transfo), path(centroids_dir), path(ref) /* optional, value = [] */
+
+    output:
+    tuple val(meta), path("*__*.trk"), emit: warped_bundle
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def reference = "$ref" ? "--reference $ref" : ""
+
+    def inverse = task.ext.inverse ? "--inverse " + task.ext.inverse : ""
+    def in_deformation = task.ext.in_deformation ? "--in_deformation " + task.ext.in_deformation : ""
+    def reverse_operation = task.ext.reverse_operation ? "--reverse_operation " + task.ext.reverse_operation : ""
+    def remove_invalid = task.ext.remove_invalid ? "--remove_invalid " + task.ext.remove_invalid : ""
+    def keep_invalid = task.ext.keep_invalid ? "--keep_invalid " + task.ext.keep_invalid : ""
+    def no_empty1 = task.ext.no_empty1 ? "--no_empty " + task.ext.no_empty1 : ""
+    def force = task.ext.force ? "-f " + task.ext.force : ""
+
+    def cut_invalid = task.ext.cut_invalid ? "--cut_invalid " + task.ext.cut_invalid : ""
+    def remove_single_point = task.ext.remove_single_point ? "--remove_single_point " + task.ext.remove_single_point : ""
+    def remove_overlapping_points = task.ext.remove_overlapping_points ? "--remove_overlapping_points " + task.ext.remove_overlapping_points : ""
+    def threshold = task.ext.threshold ? "--threshold " + task.ext.threshold : ""
+    def no_empty2 = task.ext.no_empty2 ? "--no_empty " + task.ext.no_empty2 : ""
+
+
+    """
+    for centroid in $centroids_dir/*.trk;
+        do bname=\${centroid/_centroid/}
+        bname=\$(basename \${bname} .trk)
+
+        scil_apply_transform_to_tractogram.py $centroid $anat $transfo tmp.trk\
+                        $inverse\
+                        $in_deformation\
+                        $reverse_operation\
+                        $remove_invalid\
+                        $keep_invalid\
+                        $no_empty1\
+                        $force\
+                        $reference
+
+        scil_remove_invalid_streamlines.py tmp.trk ${prefix}__${bname}.trk\
+                        $cut_invalid\
+                        $remove_single_point\
+                        $remove_overlapping_points\
+                        $threshold\
+                        $no_empty2\
+                        $reference
+    done
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        scilpy: 1.6.0
+    END_VERSIONS
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch ${prefix}__AF_L.trk
```

Comment:
As written your module will output all the centroids, not just one. Which goes back to my previous comment.

Reply:
You indeed have access to each centroid present in the directory here. Use a for loop to cycle on them and `touch` the files you need to create.